src/rpc/blockchain.cpp

```diff
@@ -2334,30 +2362,115 @@ UniValue CreateUTXOSnapshot(
         tip = CHECK_NONFATAL(chainstate.m_blockman.LookupBlockIndex(stats.hashBlock));
     }
 
-    LOG_TIME_SECONDS(strprintf("writing UTXO snapshot at height %s (%s) to file %s (via %s)",
-        tip->nHeight, tip->GetBlockHash().ToString(),
-        fs::PathToString(path), fs::PathToString(temppath)));
+    switch (format) {
+    case UTXOSnapshotFormat::COMPACT: {
+        LOG_TIME_SECONDS(strprintf("writing UTXO snapshot at height %s (%s) to file %s (via %s)",
+                                   tip->nHeight, tip->GetBlockHash().ToString(),
+                                   fs::PathToString(path), fs::PathToString(temppath)));
+        SnapshotMetadata metadata{tip->GetBlockHash(), stats.coins_count, tip->nChainTx};
 
-    SnapshotMetadata metadata{tip->GetBlockHash(), stats.coins_count, tip->nChainTx};
+        afile << metadata;
 
-    afile << metadata;
+        COutPoint key;
+        Coin coin;
+        unsigned int iter{0};
+
+        while (pcursor->Valid()) {
+            if (iter % 5000 == 0) node.rpc_interruption_point();
+            ++iter;
+            if (pcursor->GetKey(key) && pcursor->GetValue(coin)) {
+                afile << key;
+                afile << coin;
+            }
 
-    COutPoint key;
-    Coin coin;
-    unsigned int iter{0};
-
-    while (pcursor->Valid()) {
-        if (iter % 5000 == 0) node.rpc_interruption_point();
-        ++iter;
-        if (pcursor->GetKey(key) && pcursor->GetValue(coin)) {
-            afile << key;
-            afile << coin;
+            pcursor->Next();
         }
 
-        pcursor->Next();
+        afile.fclose();
+        break;
     }
+    case UTXOSnapshotFormat::SQLITE: {
+#ifdef USE_SQLITE
+        LOG_TIME_SECONDS(strprintf("writing UTXO snapshot at height %s (%s) to sqlite DB file %s",
+                                   tip->nHeight, tip->GetBlockHash().ToString(), fs::PathToString(path)));
+
+        sqlite3* db;
+        int ret;
+
+        if (!ValidAsCString(fs::PathToString(path))) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid path");
+        }
+
+        ret = sqlite3_open(fs::PathToString(path).c_str(), &db);
+        if (ret != SQLITE_OK) {
+            sqlite3_close(db);
+            throw JSONRPCError(RPC_INTERNAL_ERROR, strprintf("Unable to open database file %s", path.u8string()));
```

Comment:
maybe we could cover this error in the functional test, passing a wrong path.

Reply:
Thanks. Have added one!